app/Http/Controllers/SectorController.php

```diff
@@ -57,35 +66,88 @@ public function store(Request $request)
     public function show(string $id)
     {
         $sector = Sector::findOrFail($id);
-        return view('ledger.show', [
+        return view('sectors.show', [
             'sector' => $sector
         ]);
     }
 
     /**
      * Show the form for editing the specified resource.
      */
-    public function edit(string $id)
+    public function edit(Request $request, string $id)
     {
-        $sector = Sector::findOrFail($id);
-        return view('sectors.edit', [
-            'sector' => $sector
-        ]);
+        if($request->user()->isAdmin())
+        {
+            $sector = Sector::findOrFail($id);
+            return view('sectors.edit', [
+                'sector' => $sector
+            ]);
+        }
+        else
+        {
+            abort(401);
+        }
     }
 
     /**
      * Update the specified resource in storage.
      */
     public function update(Request $request, string $id)
     {
-        //
+        // Validate the incoming request data
+        $validated = $request->validate([
+            'name' => 'required|string|max:255',               // Validate name (required, string, max 255 characters)
+            'url' => 'nullable|string|max:2555',
+            'description' => 'nullable|string|max:255'
+        ]);
+
+        // Find the department by ID
+        $sector = Sector::findOrFail($id);
+
+        // Update the department profile with the validated data
+        $sector->update($validated);
+
+        // Optionally, flash a success message to the session
+        return redirect()->route('sectors.index')
+            ->with('success', [
+                'message' => "Sector <span class=\"text-brand-green\">{$sector->name}</span> updated successfully",
+                'action_text' => 'View Sector',
+                'action_url' => route('sectors.show', $sector->id),
+            ]);
+    }
+
+    public function delete(Request $request, string $id)
+    {
+        if($request->user()->isAdmin())
+        {
+            $sector = Sector::findOrFail($id);
+            return view('sectors.delete_confirm', [
+                'sector' => $sector
+            ]);
+        }
+        else
+        {
+            abort(401);
+        }
     }
 
     /**
      * Remove the specified resource from storage.
      */
-    public function destroy(string $id)
+    public function destroy(Request $request, string $id)
     {
-        //
+        if($request->user()->isAdmin())
```

Comment:
Refactor to use `Auth::user`

Reply:
resolved in 59ff09f7743b739de37cce6a45b1e3a7053d17fb